### PR TITLE
fix: no error when loading undefined schedule name

### LINKED ISSUE #554: Website crashes when editing schedule too quickly
If you open the website and immediately try to click the Edit Schedule button, the website crashes because `scheduleName` is undefined.

We resolved a similar issue here: #534 

https://user-images.githubusercontent.com/78244965/232614127-6d00cf20-35bc-484e-927f-5fbb7467fef1.mp4

### PR DESCRIPTION
**Pull request by ap0nia**
## Summary
The schedule store's `getScheduleName` can access a schedule at an undefined index in the schedules array.

Closes #554

### CODE REVIEW

File: apps/antalmanac/src/lib/course-helpers.ts
Comment (ap0nia) [inline apps/antalmanac/src/lib/course-helpers.ts:1]: Some dipshit setup a circular dependency between the AppStore and ScheduleStores, so they're imported in the wrong order somehow; I can't do any testing for schedule store unless I fix it.
image

**Comment by KevinWu098** [comment]
I believe this should close #554

File: apps/antalmanac/src/stores/Schedules.ts [hunk 1]
```diff
@@ -50,7 +51,7 @@ export class Schedules {
      * @return a specific schedule name
      */
     getScheduleName(scheduleIndex: number) {
-        return this.schedules[scheduleIndex].scheduleName;
+        return this.schedules[scheduleIndex]?.scheduleName;
```
Comment (EricPedley) [inline apps/antalmanac/src/stores/Schedules.ts:54]: This is is the actual fix right? Everything else was just a refactor to make writing the test work?

Comment (ap0nia) [reply apps/antalmanac/src/stores/Schedules.ts:54]: Yes, I just wanted to add a question mark and a test for that lol

**Review by EricPedley — APPROVED** [review]

**Pull request merged**